### PR TITLE
Added JUnit tests for Commons package

### PR DESCRIPTION
**Pull request by code-incubus**
Signed-off-by: code-incubus nemanja.milutinovic@comtrade.com

Added JUnit tests for classes in a commons package:

- ArgumentValidator.java
- KapuaFileUtils.java
- CommonsValidationRegexTest.java

Related Issue
None

Description of the solution adopted
/
Screenshots
/

Any side note on the changes made
- I put a comment at line 82 in class KapuaFileUtils.java below the "FIXME". We need to adjust the tests after the changes in this class.

### CODE REVIEW

File: commons/src/test/java/org/eclipse/kapua/commons/util/KapuaFileUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        String[] permittedFormats = new String[] {"https://www.example3345.com/", "http://www.example.com:1080/docs/resource1.html",
+                "https://www.example.com:1080/docs/resource2.html", "file:///c:/EXAMPLE/clock.example"};
+        String [] invalidFormats = new String [] {"www.example.com:1080/docs/resource1.html", "/c:/WINDOWS/clock.example",
+                "http://www.example.com:10800000000000/docs/resource1.html", "htp://www.example.com:1080/docs/resource1.html"};
+
+        URL[] expectedUrl = new URL[] {new URL("https://www.example3345.com/"), new URL("http://www.example.com:1080/docs/resource1.html"),
+                                       new URL("https://www.example.com:1080/docs/resource2.html"), new URL("file:///c:/EXAMPLE/clock.example")};
+        for (int i = 0; i < permittedFormats.length; i++) {
+            try {
+                assertEquals(expectedUrl[i], KapuaFileUtils.getAsURL(permittedFormats[i]));
```
Comment (Coduz) [inline commons/src/test/java/org/eclipse/kapua/commons/util/KapuaFileUtilsTest.java]: assertEquals has an overload method that accepts a message as a first argument. So `fail` invoked in the catch exception is not the correct usage of `Assert` stuff. 
Please change this and also other similar usages.

File: commons/src/test/java/org/eclipse/kapua/commons/util/KapuaFileUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            } catch (Exception ex) {
+                fail("Inappropriate http scheme.");
+            }
+            try {
+                assertEquals(invalidFormats[i], KapuaFileUtils.getAsURL(invalidFormats[i]));
```
Comment (Coduz) [inline commons/src/test/java/org/eclipse/kapua/commons/util/KapuaFileUtilsTest.java]: If they are not intended to match please use `assertNotEquals`. 
If `KapuaFileUtils.getAsURL(...)` is meant to throw an Exception catch the Exception and do asserts on the type of Exception expected.

**Review by Coduz — CHANGES_REQUESTED** [review]
Hi @code-incubus ,

I've made two comments but those should be changed everywhere the same this is done.

Thanks,

_Alberto_

**Comment by code-incubus** [comment]
Hi @Coduz,
I made all the changes that you mentioned above. If something is not ok please let me know.
I'll fix it up quickly.

**Review by LeoNerdoG — APPROVED** [review]
Hey @Coduz,
now this PR looks ok, I think this can be merged.

File: commons/src/main/java/org/eclipse/kapua/commons/util/KapuaFileUtils.java [hunk 3]
```diff
@@ -79,6 +79,7 @@ public static File getAsFile(String filePath) throws KapuaSettingException {
 
             if (hasHttpScheme(filePath)) {
                 // FIXME:
+                // After fixing this method, needs to adjust/fix the tests
```
Comment (Coduz) [inline commons/src/main/java/org/eclipse/kapua/commons/util/KapuaFileUtils.java]: Which is the issue with this method?

Comment (code-incubus) [reply commons/src/main/java/org/eclipse/kapua/commons/util/KapuaFileUtils.java]: @Coduz 
That’s the comment I found in that class. So if that method ever changes or whatever ... it is necessary to adjust the test. That's all.

**Review by LeoNerdoG — APPROVED** [review]

**Review by Coduz — APPROVED** [review]

**Pull request merged**